src/resolve/resolvectl.c

```diff
@@ -476,7 +476,7 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         log_debug("Resolving %s %s %s (interface %s).", name, dns_class_to_string(class), dns_type_to_string(type), isempty(arg_ifname) ? "*" : arg_ifname);
 
-        if (single_label_nonsynthetic(name))
+        if (!endswith(name, ".") && single_label_nonsynthetic(name))
```

Comment:
please use dns_name_dot_suffixed() for correctness (i.e. the trailing dot might be escaped in which case it isn't actually a dot)

Reply:
Thanks, updated.